source/languageDetection.py

```diff
@@ -0,0 +1,284 @@
+#A part of NonVisual Desktop Access (NVDA)
+#Copyright (C) 2015 Dinesh Kaushal, NV Access Limited
+#This file is covered by the GNU General Public License.
+#See the file COPYING for more details.
+
+
+
+"""Handles detection of languages in Unicode text.
+"""
+
+import speech
+from speech import SpeechCommand
+from speech import LangChangeCommand
+import languageHandler
+import config
+from unicodeScriptData import scriptCode
+from collections import OrderedDict
+from logHandler import log
+import unicodedata
+
+# maintains list of priority languages as a list of languageID, ScriptName, and LanguageDescription
+languagePriorityListSpec = []
+
+scriptIDToLangID = {}
```

Comment:
doc string? what is this used for, why does it exist when `langIDtoScriptID` exists?

Reply:
scriptIDToLangID is reverse of langIDToScriptID and is used to obtain language of the current script. language of a script is used to detect whether a chunk should be broken for languages that use multiple scripts.

Reply:
added the doc string